src/main/kotlin/me/ebonjaeger/perworldinventory/conversion/ConvertTask.kt

```diff
@@ -0,0 +1,66 @@
+package me.ebonjaeger.perworldinventory.conversion
+
+import org.bukkit.ChatColor
+import org.bukkit.OfflinePlayer
+import org.bukkit.command.CommandSender
+import org.bukkit.scheduler.BukkitRunnable
+
+/**
+ * Task to convert player data from MultiVerse-Inventories to PWI.
+ *
+ * @property convertService The [ConvertService] running this task.
+ * @property sender The [CommandSender] that started the conversion.
+ * @property offlinePlayers All [OfflinePlayer]s on the server.
+ */
+class ConvertTask (private val convertService: ConvertService,
+                   private val sender: CommandSender,
+                   private val offlinePlayers: Array<out OfflinePlayer>) : BukkitRunnable()
+{
+
+    private val CONVERTS_PER_TICK = 5
+
+    private var currentPage = 0
+
+    override fun run()
+    {
+        val stopIndex = currentPage * CONVERTS_PER_TICK + CONVERTS_PER_TICK
+        var currentIndex = currentPage * CONVERTS_PER_TICK
+
+        if (currentIndex >= offlinePlayers.size)
+        {
+            finish()
+            return
+        }
+
+        val playersInPage = mutableSetOf<OfflinePlayer>()
+        while (currentIndex < stopIndex && currentIndex < offlinePlayers.size)
+        {
+            playersInPage.add(offlinePlayers[currentIndex])
+            currentIndex++
+        }
+
+        currentPage++
+
+        convertService.executeConvert(playersInPage)
+        if (currentPage % 20 == 0)
+        {
+            sender.sendMessage("${ChatColor.BLUE}» ${ChatColor.GRAY}Convert progress: " +
+                    "$stopIndex/${offlinePlayers.size}")
+        }
```

Comment:
This is very clean logic. Easy conditions, nice intuitive logging. Good job!

Reply:
:D Thanks!